pkg/daemon/config.go

```diff
@@ -107,43 +107,45 @@ func tryToLoadOldConfig(nodeProfilesJson []byte) ([]ptpv1.PtpProfile, bool) {
 }
 
 func (output *ptp4lConf) populatePtp4lConf(config *string) error {
-	lines := strings.Split(*config, "\n")
 	var currentSectionName string
 	var currentSection ptp4lConfSection
 	output.sections = make([]ptp4lConfSection, 0)
 	globalIsDefined := false
 
-	for _, line := range lines {
-		line = strings.TrimSpace(line)
-		if strings.HasPrefix(line, "#") {
-			continue
-		} else if strings.HasPrefix(line, "[") {
-			if currentSectionName != "" {
-				output.sections = append(output.sections, currentSection)
+	if config != nil {
+		for _, line := range strings.Split(*config, "\n") {
+			glog.Infof(line)
+			line = strings.TrimSpace(line)
+			if strings.HasPrefix(line, "#") {
+				continue
+			} else if strings.HasPrefix(line, "[") {
+				if currentSectionName != "" {
+					output.sections = append(output.sections, currentSection)
+				}
+				currentSectionName = line
+				currentLine := strings.Split(line, "]")
+
+				if len(currentLine) < 2 {
+					return errors.New("Section missing closing ']'")
```

Comment:
include value of `currentLine` would help debug

Reply:
updated